### PR TITLE
fix(cometbft) Prevent panic on non-nil proposal without vote extensions

### PR DESCRIPTION
**Pull request by chillyvee**
## Description

We saw that gaiad has adoped cometbft v0.38 "HEAD" but we are aware of a panic bug affecting validator nodes.

We first experienced daemon panics on seda chain due to cometbft v0.38.6 calling panic when a non-nil proposal vote has vote extensions disabled. 

The panic is a mistake and cometbft v0.38.11 includes our fix.

This panic is likely to affect gaiad regardless of vote extensions being enabled or not since the bug is:
```
extSignature = len([]byte{}) > 0 = false
precommit = true
isNil = false

if extSignature == (!precommit || isNil) => if faise == (!true || false) => if false == (false || false) => if false == false {
 panic()
}
```

Link to cometbft PR and discussion:
https://github.com/cometbft/cometbft/pull/3565

This fix can reduce gaiad panics on upcoming mainnet v19

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [X] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [X] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#changelog))
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: .changelog/unreleased/bug-fixes/3270-no-panic-voteext.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- `[cometbft]` Prevent panic on non-nil Precommit when vote extensions are enabled
+  cometbft ([\#3270](https://github.com/cosmos/gaia/pull/3270)
```
Comment (mpoke) [inline .changelog/unreleased/bug-fixes/3270-no-panic-voteext.md]: This PR bumps the version of CometBFT. The bug was fixed in the comet repo and it's marked in their changelogs. 
```suggestion
- Bump [cometbft](https://github.com/cometbft/cometbft) to
   [v0.38.11](https://github.com/cometbft/cometbft/releases/tag/v0.38.11)
   ([\#3270](https://github.com/cosmos/gaia/pull/3270)
```

File: .changelog/unreleased/bug-fixes/3270-no-panic-voteext.md
Comment (mpoke) [inline .changelog/unreleased/bug-fixes/3270-no-panic-voteext.md:1]: Rename the file to `3270-bump-cometbft` and move it to `.changelog/unreleased/dependencies/`.

**Review by mpoke — COMMENTED** [review]
@chillyvee Thanks for opening the PR. See my comments below. Also, is this change state-breaking? If so, please add the changelog entry also to the `.changelog/unreleased/state-breaking` folder.

**Comment by chillyvee** [comment]
> Also, is this change state-breaking?

From what we know, this is not state breaking.  We mixed this fix into seda without updating other nodes and remain in consensus without panics.  The bug fix only removes an accidental panic preventing normal program flow.

Updated changelog as you suggest

**Review by mpoke — DISMISSED** [review]
@chillyvee could you please rebase on main? I added the v19.x backport label so it will be automatically backported to release/v19.x

**Comment by chillyvee** [comment]
> @chillyvee could you please rebase on main? I added the v19.x backport label so it will be automatically backported to release/v19.x

PR rebased on main.

**Comment by sainoe** [comment]
LGTM. Thanks @chillyvee!

**Review by sainoe — APPROVED** [review]

**Review by mpoke — APPROVED** [review]

**Pull request merged**